packages/client/src/runtime/utils/find.ts

```diff
@@ -68,36 +71,45 @@ export function findSync(
   try {
     const realRoot = realpathSync(root)
 
-    // we make sure not to loop infinitely
-    if (seen[realRoot]) {
-      return found
-    }
-
-    // we stop if we found enough results
-    if (limit - found.length <= 0) {
-      return found
-    }
-
     // we check that the root is a directory
     if (direntToType(statSync(realRoot)) !== 'd') {
       return found
     }
 
-    // we list the items in the current root
-    const items = readdirSync(root, { withFileTypes: true })
+    // we list the items in the search root
+    const items = readdirSync(realRoot, { withFileTypes: true })
+    const stack = [...items.map((i) => ({ item: i, dir: root }))]
```

Comment:
nit: technically it's a queue, not a stack, a stack would lead to depth-first search

Reply:
hilarious, fixed